File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n=len(height)
        maximum=[]
        left_index=0
        right_index=n-1
        while left_index<right_index:
            h=min([height[left_index],height[right_index]])
            w=right_index-left_index
            dist=h*w
            maximum.append(dist)
            if height[left_index]<height[right_index]:
                left_index+=1
            else:
                right_index-=1
        return max(maximum)

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int maxwater=0;
        while(left<right){
            int h=Math.min(height[left],height[right]);
            int w=right-left;
            int a=h*w;
            maxwater=Math.max(maxwater,a);

            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxwater;
    }
}